src/ml/neural_net/style_transfer/mps_style_transfer_utils.m

```diff
@@ -0,0 +1,657 @@
+  transformerDesc.decode1.conv.updateWeights = tuneAllWeights;
+
+  // Decode 1 Instance Normalization
+
+  transformerDesc.decode1.inst.channels = 64;
+  transformerDesc.decode1.inst.styles = numStyles;
+  transformerDesc.decode1.inst.label = @"transformer_decode_1_inst";
+
+  // Decode 1 Upsampling
+
+  transformerDesc.decode1.upsample.scale = 2;
+
+  // Decode 2 Convolution
+
+  transformerDesc.decode2.conv.kernelWidth = 3;
+  transformerDesc.decode2.conv.kernelHeight = 3;
+  transformerDesc.decode2.conv.inputFeatureChannels = 64;
+  transformerDesc.decode2.conv.outputFeatureChannels = 32;
+  transformerDesc.decode2.conv.strideWidth = 1;
+  transformerDesc.decode2.conv.strideHeight = 1;
+  transformerDesc.decode2.conv.paddingWidth = 1;
+  transformerDesc.decode2.conv.paddingHeight = 1;
+  transformerDesc.decode2.conv.label = @"transformer_decode_2_conv";
+  transformerDesc.decode2.conv.updateWeights = tuneAllWeights;
+
+  // Decode 2 Instance Normalization
+
+  transformerDesc.decode1.inst.channels = 32;
+  transformerDesc.decode1.inst.styles = numStyles;
+  transformerDesc.decode1.inst.label = @"transformer_decode_2_inst";
+
+  // Decode 2 Upsampling
+
+  transformerDesc.decode1.upsample.scale = 2;
+
+  // Decode 3 Convolution
+
+  transformerDesc.conv.kernelWidth = 9;
+  transformerDesc.conv.kernelHeight = 9;
+  transformerDesc.conv.inputFeatureChannels = 32;
+  transformerDesc.conv.outputFeatureChannels = 3;
+  transformerDesc.conv.strideWidth = 1;
+  transformerDesc.conv.strideHeight = 1;
+  transformerDesc.conv.paddingWidth = 1;
+  transformerDesc.conv.paddingHeight = 1;
+  transformerDesc.conv.label = @"transformer_decode_3_conv";
+  transformerDesc.conv.updateWeights = tuneAllWeights;
+
+  // Decode 3 Instance Normalization
+
+  transformerDesc.decode1.inst.channels = 3;
+  transformerDesc.decode1.inst.styles = numStyles;
+  transformerDesc.decode1.inst.label = @"transformer_decode_3_inst";
+
+  return transformerDesc;
+}
+/*
+  self.conv1_1 = _nn.Conv2D(in_channels=3, channels=64, kernel_size=3, padding=1)
+  self.conv1_2 = _nn.Conv2D(in_channels=64, channels=64, kernel_size=3, padding=1)
+
+  self.conv2_1 = _nn.Conv2D(in_channels=64, channels=128, kernel_size=3, padding=1)
+  self.conv2_2 = _nn.Conv2D(in_channels=128, channels=128, kernel_size=3, padding=1)
+
+  self.conv3_1 = _nn.Conv2D(in_channels=128, channels=256, kernel_size=3, padding=1)
+  self.conv3_2 = _nn.Conv2D(in_channels=256, channels=256, kernel_size=3, padding=1)
+  self.conv3_3 = _nn.Conv2D(in_channels=256, channels=256, kernel_size=3, padding=1)
+
+  self.conv4_1 = _nn.Conv2D(in_channels=256, channels=512, kernel_size=3, padding=1)
+  self.conv4_2 = _nn.Conv2D(in_channels=512, channels=512, kernel_size=3, padding=1)
+  self.conv4_3 = _nn.Conv2D(in_channels=512, channels=512, kernel_size=3, padding=1)
+*/
+
++ (TCMPSVgg16Descriptor *) defineVGG16Descriptor:(NSUInteger)numStyles {
+  TCMPSVgg16Descriptor* vgg16Desc;
```

Comment:
Another case where you need to actually create an instance. ObjC pointers are like C and C++ pointers: by default they are just undefined values (invalid pointers)

Reply:
Got it! Made sure to allocate instances to pointer I had defined!